packages/fuel-indexer-lib/src/manifest.rs

```diff
@@ -61,19 +86,60 @@ pub struct Manifest {
     pub graphql_schema: String,
     pub module: Module,
     pub metrics: Option<bool>,
-    pub contract_id: Option<String>,
+    #[serde(
+        deserialize_with = "contract_id_from_str",
+        serialize_with = "contract_id_to_str"
+    )]
+    pub contract_id: ContractIds,
     pub start_block: Option<u64>,
     #[serde(default)]
     pub resumable: Option<bool>,
 }
 
+fn contract_id_from_str<'de, D>(deserializer: D) -> Result<ContractIds, D::Error>
+where
+    D: serde::Deserializer<'de>,
+{
+    let value = serde_yaml::Value::deserialize(deserializer)?;
+    match value {
+        serde_yaml::Value::String(s) => {
+            ContractIds::from_str(&s).map_err(serde::de::Error::custom)
+        }
+        serde_yaml::Value::Sequence(seq) => {
+            let ids = seq
+                .into_iter()
+                .filter_map(|val| match val {
+                    serde_yaml::Value::String(s) => Some(s),
+                    _ => None,
+                })
+                .collect::<Vec<_>>();
+            Ok(ContractIds::Multiple(ids.into_iter().map(Some).collect()))
+        }
+        serde_yaml::Value::Null => Ok(ContractIds::Single(None)),
+        _ => Err(serde::de::Error::custom("Invalid contract_id value")),
```

Comment:
Looks like we might be able to integrate `serve::de::Error` into `ManifestError`?

Reply:
I tried this for a while as I really wanted this. But because we are doing this funky this with serde in the `Manifest` struct :
```rust
#[serde(
        serialize_with = "ContractIds::serialize",
        deserialize_with = "ContractIds::deserialize"
    )]
    pub contract_id: ContractIds,
```
It always needs to return it's own very specific error and complains when I try to return `ManifestError`, I tried to get around this by tying those types together, but in short it got messy and wasn't working.